### PR TITLE
fix(cli): UnknownError when upgrading

### PR DESCRIPTION
**Pull request by a1exwang**
fix: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/12684229/

### CODE REVIEW

**Comment by a1exwang** [comment]
E2E TEST: https://github.com/OfficeDev/TeamsFx/blob/dev/packages/cli/tests/e2e/multienv/TestRemoteHappyPath.tests.ts

File: packages/fx-core/src/core/middleware/projectMigrator.ts [hunk 1]
```diff
@@ -177,6 +177,8 @@ export const ProjectMigratorMW: Middleware = async (ctx: CoreHookContext, next:
 
     try {
       await migrateToArmAndMultiEnv(ctx);
+      // return ok for the lifecycle functions to prevent breaking error handling logic.
+      ctx.result = ok({});
```
Comment (meifans) [inline packages/fx-core/src/core/middleware/projectMigrator.ts:181]: Maybe `undefined` is better.
```suggestion
      ctx.result = ok(undefined);
```

**Review by meifans — APPROVED** [review]
LGTM

File: packages/fx-core/src/core/middleware/projectMigrator.ts (hunk 1 above)
Comment (a1exwang) [reply packages/fx-core/src/core/middleware/projectMigrator.ts:181]: The method signature is `Result<Void, FxError>`, Void is {}, so I use {}

**Review by qinezh — APPROVED** [review]

**Review by meifans — APPROVED** [review]

**Review by dooriya — APPROVED** [review]

**Pull request merged**